File: Serene1.Web/Modules/MovieDB/Movie/MovieForm.cs
using MovieTutorial.MovieDB;
using Serenity.ComponentModel;
using System;

namespace Serene1.MovieDB.Forms;

[FormScript("MovieDB.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieForm
{
    public string Title { get; set; }
    public string Description { get; set; }
    public string Storyline { get; set; }
    public int Year { get; set; }
    public DateTime ReleaseDate { get; set; }
    public int GenreId { get; set; }
    public int Runtime { get; set; }
    public MovieKind Kind { get; set; }
}

File: Serene1.Web/Modules/ServerTypes/MovieDB/MovieForm.ts
import { StringEditor, IntegerEditor, DateEditor, EnumEditor, PrefixedContext } from "@serenity-is/corelib";
import { MovieKind } from "../MovieTutorial/MovieDB.MovieKind";
import { initFormType } from "@serenity-is/corelib/q";

export interface MovieForm {
    Title: StringEditor;
    Description: StringEditor;
    Storyline: StringEditor;
    Year: IntegerEditor;
    ReleaseDate: DateEditor;
    GenreId: IntegerEditor;
    Runtime: IntegerEditor;
    Kind: EnumEditor;
}

export class MovieForm extends PrefixedContext {
    static formKey = 'MovieDB.Movie';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MovieForm.init)  {
            MovieForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = DateEditor;
            var w3 = EnumEditor;

            initFormType(MovieForm, [
                'Title', w0,
                'Description', w0,
                'Storyline', w0,
                'Year', w1,
                'ReleaseDate', w2,
                'GenreId', w1,
                'Runtime', w1,
                'Kind', w3
            ]);
        }
    }
}

[MovieKind]; // referenced types